drivers/wireless/ieee80211/bcm43xxx/bcmf_netdev.c

```diff
@@ -755,6 +770,74 @@ static int bcmf_ifdown(FAR struct net_driver_s *dev)
   return OK;
 }
 
+/****************************************************************************
+ * Name: bcmf_lowpower_work
+ *
+ * Description:
+ *   Process low power saving dueto work timer expiration
+ *
+ * Input Parameters:
+ *   arg - context of device to use
+ *
+ ****************************************************************************/
+
+#ifdef CONFIG_IEEE80211_BROADCOM_LOWPOWER
+static void bcmf_lowpower_work(FAR void *arg)
+{
+  FAR struct bcmf_dev_s *priv = (FAR struct bcmf_dev_s *)arg;
+  clock_t cticks = clock_systime_ticks();
+
+  if (!priv->bc_bifup)
+    {
+      return;
+    }
+
+  if (TICK2SEC(cticks - priv->lp_time) >=
+      CONFIG_IEEE80211_BROADCOM_LOWPOWER_TIMEOUT)
+    {
+      bcmf_wl_set_pm(priv, PM_MAX);
+      priv->lp_pending = false;
+    }
+  else
+    {
+      cticks = SEC2TICK(CONFIG_IEEE80211_BROADCOM_LOWPOWER_TIMEOUT) -
```

Comment:
let's calculate the delta before if, so we can reuse the result at here and line 795